src/hooks/use-chat-context.tsx

```diff
@@ -0,0 +1,183 @@
+}
+
+const parseWpCliOutput = ( stdout: string, defaultValue: string[] ): string[] => {
+	try {
+		const data = JSON.parse( stdout );
+		return data?.map( ( item: { name: string } ) => item.name ) || [];
+	} catch ( error ) {
+		console.error( error );
+	}
+	return defaultValue;
+};
+
+export const ChatProvider: React.FC< ChatProviderProps > = ( { children } ) => {
+	const [ initialLoad, setInitialLoad ] = useState< Record< string, boolean > >( {} );
+	const installedApps = useCheckInstalledApps();
+	const { data: sites, loadingSites, selectedSite } = useSiteDetails();
+	const wpVersion = useGetWpVersion( selectedSite || ( {} as SiteDetails ) );
+	const [ pluginsList, setPluginsList ] = useState< Record< string, string[] > >( {} );
+	const [ themesList, setThemesList ] = useState< Record< string, string[] > >( {} );
+	const numberOfSites = sites?.length || 0;
+	const sitePath = selectedSite?.path || '';
+	const sitePort = selectedSite?.port || '';
+	const siteThemeDetails = selectedSite?.themeDetails;
+
+	const availableEditors = Object.keys( installedApps ).filter( ( app ) => {
+		return installedApps[ app as keyof InstalledApps ];
+	} );
+
+	const fetchPluginList = useCallback( async ( path: string ) => {
+		const { stdout, stderr } = await getIpcApi().executeWPCLiInline( {
+			projectPath: path,
+			args: 'plugin list --format=json --status=active',
+		} );
+		if ( stderr ) {
+			return [];
+		}
+		return parseWpCliOutput( stdout, [] );
+	}, [] );
+
+	const fetchThemeList = useCallback( async ( path: string ) => {
+		const { stdout, stderr } = await getIpcApi().executeWPCLiInline( {
+			projectPath: path,
+			args: 'theme list --format=json',
+		} );
+		if ( stderr ) {
+			return [];
+		}
+		return parseWpCliOutput( stdout, [] );
+	}, [] );
+
+	useEffect( () => {
+		let isCurrent = true;
+		// Initial load. Prefetch all the plugins and themes for the sites.
+		const run = async () => {
+			const result = await Promise.all( [
+				fetchPluginList( sitePath ),
+				fetchThemeList( sitePath ),
+			] );
+			if ( isCurrent && selectedSite?.id ) {
+				setInitialLoad( ( prev ) => ( { ...prev, [ selectedSite.id ]: true } ) );
+				setPluginsList( ( prev ) => ( { ...prev, [ selectedSite.id ]: result[ 0 ] } ) );
+				setThemesList( ( prev ) => ( { ...prev, [ selectedSite.id ]: result[ 1 ] } ) );
+			}
+		};
+		if (
+			selectedSite &&
+			! loadingSites &&
+			! initialLoad[ selectedSite.id ] &&
+			isCurrent &&
+			! pluginsList[ selectedSite.id ] &&
+			! themesList[ selectedSite.id ]
+		) {
+			run();
+		}
+		return () => {
+			isCurrent = false;
+		};
+	}, [
+		fetchPluginList,
+		fetchThemeList,
+		initialLoad,
+		loadingSites,
+		pluginsList,
+		selectedSite,
+		sites,
+		themesList,
+		sitePath,
+	] );
+
+	useWindowListener( 'focus', async () => {
+		// When the window is focused, we need to kick off a request to refetch the theme details, if server is running.
+		if ( ! selectedSite?.id || selectedSite.running === false ) {
```

Comment:
In theory, plugins and themes can be obtained using `wp-cli` even when the site is not running. What do you think about removing that condition?

Reply:
Hey @fluiddot! 
Initially, the `useEffect` fetches plugins and themes. Even if the user focus out, and back in, there is no possibility for the user to have changed his theme, correct?
Given that, it's not necessary to refetch everything, if the site is not running

Reply:
> Initially, the `useEffect` fetches plugins and themes. Even if the user focus out, and back in, there is no possibility for the user to have changed his theme, correct?
> Given that, it's not necessary to refetch everything, if the site is not running

Ah, good point. Although, in theory, it would be possible to change themes and plugins via `wp-cli` which doesn't require starting the site. This might be an edge case, so we could leave this as-is.